packages/nimble-components/src/wafer-map/modules/computations.module.ts

```diff
@@ -0,0 +1,189 @@
+import { scaleBand, ScaleLinear, scaleLinear } from 'd3-scale';
+import type { WaferMapDie } from '../types';
+import { Dimensions, Margin, WaferMapQuadrant } from '../types';
+
+interface MapDimensions {
+    origin: {
+        x: number,
+        y: number
+    };
+    rows: number;
+    cols: number;
+}
+
+/**
+ * Computations module
+ */
+export class Computations {
+    public containerDimensions: Dimensions;
+    public dieDimensions: Dimensions = { width: 0, height: 0 };
+    public radius: number;
+    public margin: Margin;
+
+    public horizontalScale: ScaleLinear<number, number>;
+    public verticalScale: ScaleLinear<number, number>;
```

Comment:
It looks like the goal is to calculate these properties within the constructor, should these be marked readonly so external code can't accidentally mutate the values?

```suggestion
    public readonly containerDimensions: Dimensions;
    public readonly dieDimensions: Dimensions = { width: 0, height: 0 };
    public readonly radius: number;
    public readonly margin: Margin;

    public readonly horizontalScale: ScaleLinear<number, number>;
    public readonly verticalScale: ScaleLinear<number, number>;
```